### ISSUE TITLE
Migrate ClusterLoader2 from text/template to github.com/google/safetext/yamltemplate

### ISSUE THREAD

**Issue by jprzychodzen** [kubernetes/perf-tests#2245]
ClusterLoader2 is using unsecure `text/template`. We should migrate to use safer version - `github.com/google/safetext/yamltemplate`.

/good-first-issue

**Comment by Sajiyah-Salat** [comment]
Can you please specify this problem in brief. I am beginner trying to understand the issue. Please guide me to the file which needs to be change. what change you would like to see and what's the issue. It will help me get a clear idea.

**Comment by mborsz** [comment]
I'm afraid simple changing library isn't possible. @jupblb was already there with https://github.com/kubernetes/perf-tests/pull/2216 which was failing presubmit.

I think we need to understand why it was failing and how to fix this. I'm afraid that the reason might be that CL2 is doing some unsafe yaml operations which are simply prohibited by the library, but I'm just guessing.

**Comment by jprzychodzen** [comment]
> Why are we solving this issue?
To increase security of ClusterLoader2 users

>To address this issue, are there any code changes? If there are code changes, what needs to be done in the code and what places can the assignee treat as reference points?

We should change library in `clusterloader2/pkg/config/template_functions.go` from `text/template` to `github.com/google/safetext/yamltemplate`. This will probably break if done without following appropriate migration from [migration instruction](https://github.com/google/safetext/#instructions-for-texttemplate-replacements) . Example failing long from PR #2216:

```
F1214 13:29:32.465064   16085 clusterloader.go:325] Error while setting up prometheus stack: TemplateToObject error: replacing placeholders error: Invalid YAML Template
```

> Does this issue have zero to low barrier of entry?

It does not required expert K8s knowledge. It's expected from the candidate for this issue to
* have minimal K8s knowledge, understand that YAMLs represent objects in K8s API
* is able to code and compile in Golang

> How can the assignee reach out to you for help?

Please put comments in this issue.

**Comment by Sajiyah-Salat** [comment]
I understand the issue. i have done many contributions in k8s in past days. I know golang as well. However I am afraid if I cannot do that. you have explained the issue in detail. can you please let me know what will be neededin the codebase.

**Comment by samyakjain10** [comment]
@jprzychodzen I would like to work on this issue but am a bit confused on how to proceed.

**Comment by molarom** [comment]
@jprzychodzen So I started trying to replicate the error you mentioned above, but I wasn't able to do so. I tried replicating the failed tests from #2216 locally with a kind cluster (minus the large service test), but the tests seemed to run successfully. Could you provide some more insight as to where/how you're able to produce that error?

**Comment by AnjaliMishra1st** [comment]
Hi, I’d like to work on this. Could you please assign it to me?

**Comment by jprzychodzen** [comment]
@AnjaliMishra1st - it's already resolved by #2421

**Comment by AnjaliMishra1st** [comment]
Got it, thanks @jprzychodzen for letting me know 🙏.
I’ll look for another open issue or task to contribute to.
